src/modules/referral/services/cron-service.ts

```diff
@@ -3,43 +3,64 @@ import { Repository } from "typeorm";
 import { InjectRepository } from "@nestjs/typeorm";
 import { AppConfig } from "../../configuration/configuration.service";
 import { Deposit } from "../../scraper/model/deposit.entity";
-import { EnhancedCron } from "../../../utils";
-import { updateStickyReferralAddresses } from "../services/queries";
+import { EnhancedCron, wait } from "../../../utils";
+import { updateStickyReferralAddressesQuery } from "../services/queries";
+import { StickyReferralAddressesMechanism } from "src/modules/configuration";
 
 @Injectable()
 export class ReferralCronService {
   private logger = new Logger(ReferralCronService.name);
+  private semaphore = false;
 
   constructor(
     @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
     private appConfig: AppConfig,
   ) {}
 
-  // run cron every 8 minutes
-  @EnhancedCron("0 4-59/8 * * * *")
-  async refreshMaterializedViewCron() {
-    this.logger.log("start refreshMaterializedViewCron()");
-    if (this.appConfig.values.enableReferralsMaterializedViewRefresh) {
+  @EnhancedCron("0 */10 * * * *")
+  async startCrons() {
+    if (this.semaphore === true) {
```

Comment:
this could be condensed to one line right? `if (this.semaphore) return`

Reply:
Yes